### ISSUE TITLE
[Reveal] - Use of `@include foundation-reveal` places code at the top of stylesheet

### ISSUE THREAD

**Issue by brettsmason** [zurb/foundation-sites#10399]
When using the reveal mixin to add the styles to my main stylesheet, its ignoring the position in the stylesheet and getting added to the top.

The code is:
```
@media print, screen and (min-width: 40em) {
	.reveal, .reveal.tiny, .reveal.small, .reveal.large {
		right: auto;
		left: auto;
		margin: 0 auto;
	}
}
```
This looks related to [this commit](https://github.com/zurb/foundation-sites/commit/deaada3afd45d4e8d7d87f5115a5ef0206808fbb). 

@kball is there any other way we could solve the above without using `@extend`? I think this is what's causing the issue.

**Comment by kball** [comment]
Ah interesting... I'm not finding the original issue at the moment but that commit was attempting to address a bug with using the `reveal-modal-width` mixin (which depends on that extend) outside of the default `foundation-reveal` mixin (which previously was the only place the extend was defined).

It sounds like changing it from extend is probably a good approach... my sense is all of the reveal scss could use a pass, as it's not really very well set up for use as mixins (see https://github.com/zurb/foundation-sites/issues/10105)

In terms of exactly how to do this, I think you're better at that type of problem than I am... do you want to take a stab at it?

**Comment by brettsmason** [comment]
Yep I can take a look at this next week. Like you said probably needs a rework of the codebase.

**Comment by DanielRuf** [comment]
When I switched to a CSS version of normalize I did not get this anymore.

**Comment by DanielRuf** [comment]
Related issues
https://github.com/zurb/foundation-sites/pull/11106
https://github.com/zurb/foundation-sites/pull/11087
